src/Commerce/Generator/PayPal/CLI.php

```diff
@@ -0,0 +1,480 @@
+		}
+
+		return $ticket_ids;
+	}
+
+	/**
+	 * Parses, validating and sanity-checking them, the user-provided
+	 * attendee min and max values.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $assoc_args
+	 *
+	 * @return array
+	 *
+	 * @throws \WP_CLI\ExitException
+	 */
+	protected function parse_attendees_min_max( array $assoc_args ) {
+		if ( ! (
+				filter_var( $assoc_args['attendees_min'], FILTER_VALIDATE_INT )
+				&& filter_var( $assoc_args['attendees_max'], FILTER_VALIDATE_INT )
+			)
+		     || (int) $assoc_args['attendees_min'] > $assoc_args['attendees_max']
+		     || 0 > $assoc_args['attendees_min']
+		     || 0 > $assoc_args['attendees_max']
+		) {
+			WP_CLI::error( 'Attendees min and max should be positive integers that make sense' );
+		}
+
+		return array( (int) $assoc_args['attendees_min'], $assoc_args['attendees_max'] );
+	}
+
+	/**
+	 * Parse and validates the user-provided orders count.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $assoc_args
+	 *
+	 * @return int
+	 *
+	 * @throws \WP_CLI\ExitException
+	 */
+	protected function parse_count( array $assoc_args ) {
+		if ( ! filter_var( $assoc_args['count'], FILTER_VALIDATE_INT ) || (int) $assoc_args['count'] < 1 ) {
+			WP_CLI::error( 'The count parameter should be a positive integer' );
+		}
+
+		return (int) $assoc_args['count'];
+	}
+
+	/**
+	 * Parses and validate the user-provided PayPal order status.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $assoc_args
+	 *
+	 * @return string
+	 *
+	 * @throws \WP_CLI\ExitException
+	 */
+	protected function parse_order_status( array $assoc_args ) {
+		$order_status = trim( $assoc_args['order_status'] );
+
+		$supported_stati = array(
+			Tribe__Tickets__Commerce__PayPal__Stati::$completed,
+			Tribe__Tickets__Commerce__PayPal__Stati::$pending,
+			Tribe__Tickets__Commerce__PayPal__Stati::$refunded,
+			Tribe__Tickets__Commerce__PayPal__Stati::$denied,
+		);
+
+		if ( ! in_array( $order_status, $supported_stati ) ) {
+			WP_CLI::error( "The {$order_status} order status is not valid or suported" );
+		}
+
+		return $order_status;
+	}
+
+	/**
+	 * Hijack some PayPal related hooks to make all work.
+	 *
+	 * @since TBD
+	 */
+	protected function hijack_request_flow() {
+		// all transactions are valid, we are generating fake numbers
+		add_filter( 'tribe_tickets_commerce_paypal_validate_transaction', '__return_true' );
+
+		// mark all generated attendees as generated
+		add_action( 'event_tickets_tpp_attendee_created', function ( $attendee_id ) {
+			update_post_meta( $attendee_id, Meta_Keys::$generated_meta_key, 1 );
+		} );
+
+		add_filter( 'tribe_tickets_tpp_order_postarr', function ( $postarr ) {
+			$postarr['meta_input'][ Meta_Keys::$generated_meta_key ] = 1;
+
+			return $postarr;
+		} );
+
+		// no, do not send emails to the fake attendees
+		add_filter( 'tribe_tickets_tpp_send_mail', '__return_false' );
+
+		// do not `die` after generating tickets
+		add_filter( 'tribe_exit', function () {
```

Comment:
Why a function here?

Reply:
You need to return a handler, not handle itself. Like `wp_die`.